src/test/java/org/openmrs/addonindex/scheduled/FetchCoreVersionsListIT.java

```diff
@@ -0,0 +1,65 @@
+package org.openmrs.addonindex.scheduled;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.openmrs.addonindex.domain.artifactory.VersionList;
+import org.openmrs.addonindex.service.artifactory.VersionsService;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.springframework.web.client.RestTemplate;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.when;
+import static org.openmrs.addonindex.TestUtil.getFileAsString;
+
+@SpringBootTest
+@RunWith(SpringRunner.class)
+public class FetchCoreVersionsListIT {
+
+    @MockBean
+    private RestTemplateBuilder restTemplateBuilder;
+
+    @MockBean
+    private RestTemplate restTemplate;
+
+    @Autowired
+    private VersionsService versionsService;
+
+    @Autowired
+    private FetchCoreVersionsList fetchCoreVersionsList;
+
+    private VersionList versionList;
+
+    @Before
+    public void setUp() throws Exception {
+        when(restTemplateBuilder.build()).thenReturn(restTemplate);
+
+        ReflectionTestUtils.setField(fetchCoreVersionsList,
+                "url",
+                "https://openmrs.jfrog.io/openmrs/api/storage/public/org/openmrs/api/openmrs-api/");
+        ReflectionTestUtils.setField(fetchCoreVersionsList,
+                "strategy",FetchCoreVersionsList.Strategy.FETCH);
+
+    }
+
+    @Test
+    public void testCoreVersionsJsonHandle() throws Exception{
```

Comment:
The class that you're testing here is the one that has the business logic for excluding alpha/beta/SNAPSHOT/etc. Therefore your assertions in this test need to ensure that it's doing the right thing.

The correct thing to do would be to extract out a "filter" method, and to have another test class that's not an IT which has multiple @Test methods which test all the various behaviors.
The quick fix is to add more assertions to this test method to make sure you're excluding unwanted versions.